Dockerfile

```diff
@@ -50,47 +50,37 @@ RUN --mount=type=cache,sharing=locked,target=/var/cache/apt \
 FROM prepper-pcl-${WITH_PCL} AS prepper-ros2-0
 # Do nothing, ROS2 extension is not enabled
 FROM prepper-pcl-${WITH_PCL} AS prepper-ros2-1
-# Install ROS2: Setup timezone
-RUN --mount=type=cache,sharing=locked,target=/var/cache/apt \
-    echo 'Etc/UTC' > /etc/timezone && \
-    ln -s /usr/share/zoneinfo/Etc/UTC /etc/localtime && \
-    apt-get install -q -y --no-install-recommends tzdata
-
-# Install ROS2: Setup sources.list
-RUN echo "deb http://packages.ros.org/ros2/ubuntu jammy main" > /etc/apt/sources.list.d/ros2-latest.list
 
-# Install ROS2: Setup keys
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654
-
-# Install ROS2: Setup environment
-ENV LANG C.UTF-8
-ENV LC_ALL C.UTF-8
+# Install ROS2: Setup sources.list and keys
+RUN . /etc/os-release && \
+    echo "deb http://packages.ros.org/ros2/ubuntu $UBUNTU_CODENAME main" > /etc/apt/sources.list.d/ros2-latest.list && \
+    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 && \
+    apt-get update
 
+ARG ROS_DISTRO=humble
+ENV ROS_DISTRO=$ROS_DISTRO
```

Comment:
ROS based docker images often have this already set, to make it easier to source the install as a colcon underlay or when rosdep installing with package manifest that have dynamic dependencies that are conditioned on this env.

You may want to embed the check into the binary to warn users that they are attempting to run a resulting artifact against a different distro than it was compiled against to avoid linking or ABI issues.

Although that might not always be the case, particular if the RGL is statically compiled. I sometime see folks use running different distro if they are still wire compatible. So perhaps just a warning rather than a hard error?  Not sure.

Reply:
I certainly agree that it's worth having `ROS_DISTRO` defined here.
About the error message, I also referred to the build process where we want to make sure ROS is sourced (regardless of the version). We assumed that if `ROS_DISTRO` is set, ROS is also sourced ([link](https://github.com/RobotecAI/RobotecGPULidar/blob/feature/rework-dockerfile/extensions/ros2/install_deps.py#L55)) but it is not always true. I am going to modify this logic when we merge this PR.

Reply:
Perhaps you could check if the `AMENT_CURRENT_PREFIX` env is set when sourcing `/opt/ros/<distro>/setup.<shell>`? Not sure if that is the best approch.

https://github.com/ament/ament_package/blob/01592eb7c227ec12304635e530984d29055463d7/ament_package/template/prefix_level/setup.sh.in#L5

Is it expensive to fail/crach and just give a hit in the error message, or are we talking imediat segfault, thus a pre-check before startup?

Reply:
I have the build process in mind only. Before executing compilation, we check if everything is set properly and produce a user-friendly error message instead of getting errors from the compiler (to the extent we can predict). E.g. we are informing the user to source ROS2 instead of getting `rclcpp` package missing.

Reply:
Ah, I guess for user's not used to seeing CMake find errors, those can be a little vague at times.

Reply:
True, but it could also speed up debugging for new users. It is a trade-off.